simulator/mips/mips_instr.h

```diff
@@ -254,6 +255,9 @@ class MIPSInstr
         void execute_clo() { v_dst = count_zeros( ~v_src1); }
         void execute_clz() { v_dst = count_zeros(  v_src1); }
 
+        void execute_madd()  { v_dst = mips_multiplication<int32>(v_src1, v_src2); accumulating_instr = true; }
```

Comment:
I suggest to add another OperationType value to enum rather than add a Boolean. Defining `accumulating_instr` in execution seems wrong, as we may call this method on decoding stage before execution is performed.